src/services/ui/src/hooks/useSyncStatus.ts

```diff
@@ -0,0 +1,73 @@
+import { getItem, useGetItem } from "@/api";
+import { Route, urlEmbedQuery } from "@/components/Routing";
+import { useNavigate } from "react-router-dom";
+import { useQuery } from "@tanstack/react-query";
+import { useRef, useState } from "react";
+import { type SEATOOL_STATUS } from "shared-types";
+import { queryClient } from "@/router";
+import { ItemResult } from "shared-types/opensearch/main";
+import { OsUrlState } from "@/components";
+import { authorityById } from "@/utils";
+
+export type SeaStatus = (typeof SEATOOL_STATUS)[keyof typeof SEATOOL_STATUS];
+
+export const useSyncStatus = ({
+  path,
```

Comment:
Nice work creating this hook! 👏 

I did noticed that 5 of the instances that call use `useSyncSatus` pass in this for path: `path: location.state?.from ?? "/dashboard"`, and then 8 instances pass in `path: originPath ? originPath : "/dashboard"`.

Is there a way the work for defining the path could just live inside this hook, or maybe instead of doing this work in each component you could use a switch or something?